"Find References" sometimes doesn't work in TypeScript files

Attempting to run "Find References" on a token in a TypeScript file intermittently finds no matches.  When that happens, the Error Log view shows that the following error was raised in the `org.eclipse.lsp4e` plug-in:
```
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "java.util.List.iterator()" because "locs" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.iterator()" because "locs" is null
	at org.eclipse.lsp4e.operations.references.LSSearchQuery.lambda$0(LSSearchQuery.java:102)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718)
	... 6 more
```

Retrying the operation enough times may eventually succeed & find matches, in which case no new error appears in the Error Log.

I haven't tested if this occurs in filetypes other than TypeScript files.

Likely related to https://github.com/eclipse/wildwebdeveloper/issues/951

--- a/org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/references/LSFindReferences.java
+++ b/org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/references/LSFindReferences.java
@@ -28,7 +28,6 @@ import org.eclipse.lsp4e.LSPEclipseUtils;
 import org.eclipse.lsp4e.LanguageServerPlugin;
 import org.eclipse.lsp4e.LanguageServiceAccessor;
 import org.eclipse.lsp4e.ui.UI;
-import org.eclipse.lsp4j.services.LanguageServer;
 import org.eclipse.search.ui.NewSearchUI;
 import org.eclipse.ui.IEditorPart;
 import org.eclipse.ui.handlers.HandlerUtil;
@@ -55,9 +54,8 @@ public class LSFindReferences extends AbstractHandler implements IHandler {
 					if (languageServers.isEmpty()) {
 						return;
 					}
-					LanguageServer ls = languageServers.get(0);
 					try {
-						LSSearchQuery query = new LSSearchQuery(document, offset, ls);
+						LSSearchQuery query = new LSSearchQuery(document, offset, languageServers);
 						HandlerUtil.getActiveShell(event).getDisplay().asyncExec(() -> NewSearchUI.runQueryInBackground(query));
 					} catch (BadLocationException e) {
 						LanguageServerPlugin.logError(e);

--- a/org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/references/LSSearchQuery.java
+++ b/org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/references/LSSearchQuery.java
@@ -14,7 +14,6 @@
 package org.eclipse.lsp4e.operations.references;
 
 import java.util.List;
-import java.util.concurrent.CompletableFuture;
 
 import org.eclipse.core.resources.IFile;
 import org.eclipse.core.resources.IResource;
@@ -53,14 +52,12 @@ import org.eclipse.search.ui.text.Match;
 public class LSSearchQuery extends FileSearchQuery {
 
 	private final @NonNull IDocument document;
-	private final @NonNull LanguageServer languageServer;
+	private final @NonNull List<@NonNull LanguageServer> languageServers;
 	private final Position position;
 	private final String filename;
 
 	private LSSearchResult result;
 	private long startTime;
-
-	private CompletableFuture<List<? extends Location>> references;
 
 	/**
 	 * LSP search query to "Find references" from the given offset of the given
@@ -68,14 +65,14 @@ public class LSSearchQuery extends FileSearchQuery {
 	 *
 	 * @param document
 	 * @param offset
-	 * @param languageServer
+	 * @param languageServers
 	 * @throws BadLocationException
 	 */
-	public LSSearchQuery(@NonNull IDocument document, int offset, @NonNull LanguageServer languageServer)
+	public LSSearchQuery(@NonNull IDocument document, int offset, List<@NonNull LanguageServer> languageServers)
 			throws BadLocationException {
 		super("", false, false, null); //$NON-NLS-1$
 		this.document = document;
-		this.languageServer = languageServer;
+		this.languageServers = languageServers;
 		this.position = LSPEclipseUtils.toPosition(offset, document);
 		this.filename = Path.fromPortableString(LSPEclipseUtils.toUri(document).getPath()).lastSegment();
 	}
@@ -83,32 +80,34 @@ public class LSSearchQuery extends FileSearchQuery {
 	@Override
 	public IStatus run(IProgressMonitor monitor) throws OperationCanceledException {
 		startTime = System.currentTimeMillis();
-		// Cancel last references future if needed.
-		if (references != null) {
-			references.cancel(true);
-		}
 		AbstractTextSearchResult textResult = (AbstractTextSearchResult) getSearchResult();
 		textResult.removeAll();
 
 		try {
 			// Execute LSP "references" service
 			ReferenceParams params = new ReferenceParams();
-			params.setContext(new ReferenceContext(true));
+			params.setContext(new ReferenceContext(false));
 			params.setTextDocument(new TextDocumentIdentifier(LSPEclipseUtils.toUri(document).toString()));
 			params.setPosition(position);
-			languageServer.getTextDocumentService().references(params)
-					.thenAcceptAsync(locs -> {
-						// Loop for each LSP Location and convert it to Match search.
-						for (Location loc : locs) {
-							Match match = toMatch(loc);
-							if (match != null) {
-								result.addMatch(match);
-							}
+
+			for (LanguageServer languageServer : languageServers) {
+				languageServer.getTextDocumentService().references(params)
+				.thenAcceptAsync(locations -> {
+					if (locations == null) {
+						return;
+					}
+					// Loop for each LSP Location and convert it to Match search.
+					for (Location loc : locations) {
+						Match match = toMatch(loc);
+						if (match != null) {
+							result.addMatch(match);
 						}
-					}).exceptionally(e -> {
-						LanguageServerPlugin.logError(e);
-						return null;
-					});
+					}
+				}).exceptionally(e -> {
+					LanguageServerPlugin.logError(e);
+					return null;
+				});
+			}
 			return Status.OK_STATUS;
 		} catch (Exception ex) {
 			return new Status(IStatus.ERROR, LanguageServerPlugin.getDefault().getBundle().getSymbolicName(),

